test/e2e/gardener/managedseed/create_rotate_delete.go

```diff
@@ -6,246 +6,150 @@ package managedseed
-			g.Expect(err).NotTo(HaveOccurred())
-		}).Should(Succeed())
-
-		verifier := rotation.GardenletKubeconfigRotationVerifier{
-			GardenReader:                       f.GardenClient.Client(),
-			SeedReader:                         shootClient.Client(),
-			Seed:                               seed,
-			GardenletKubeconfigSecretName:      gardenletKubeconfigSecretName,
-			GardenletKubeconfigSecretNamespace: gardenletKubeconfigSecretNamespace,
-		}
 
-		By("Trigger gardenlet kubeconfig rotation by annotating Seed")
-		{
-			verifier.Before(ctx)
-			Eventually(func() error {
-				return triggerGardenletKubeconfigRotationViaSeed(ctx, f.GardenClient.Client(), seed)
-			}).Should(Succeed())
-			verifier.After(ctx, false)
-		}
-
-		By("Trigger gardenlet kubeconfig rotation by annotating ManagedSeed")
-		{
-			verifier.Before(ctx)
-			Eventually(func() error {
-				return triggerGardenletKubeconfigRotationViaManagedSeed(ctx, f.GardenClient.Client(), managedSeed)
-			}).Should(Succeed())
-			verifier.After(ctx, true)
-		}
-
-		By("Trigger gardenlet kubeconfig rotation by annotating its kubeconfig secret")
-		{
-			verifier.Before(ctx)
-			Eventually(func() error {
-				return triggerGardenletKubeconfigRotationViaSecret(ctx, shootClient.Client())
-			}).Should(Succeed())
-			verifier.After(ctx, false)
-		}
-
-		By("Trigger gardenlet kubeconfig auto-rotation by reducing kubeconfig validity")
-		{
-			By("Scale down gardenlet deployment and wait until no gardenlet pods exist anymore")
-			CEventually(ctx, func(g Gomega) []corev1.Pod {
-				deployment := &appsv1.Deployment{}
-				g.Expect(shootClient.Client().Get(ctx, client.ObjectKey{Name: "gardenlet", Namespace: "garden"}, deployment)).To(Succeed())
-
-				if ptr.Deref(deployment.Spec.Replicas, 0) != 0 {
-					By("Scale down gardenlet deployment to prevent interference of old pods with old validity settings")
-					// See https://github.com/gardener/gardener/issues/6766 for details
-
-					patch := client.MergeFrom(deployment.DeepCopy())
-					deployment.Spec.Replicas = ptr.To[int32](0)
-					g.Expect(shootClient.Client().Patch(ctx, deployment, patch)).To(Succeed())
-				}
+			Eventually(ctx, s.ShootContext.ShootKomega.Update(deployment, func() {
+				deployment.Spec.Replicas = ptr.To(int32(0))
+			})).Should(Succeed())
+		}, SpecTimeout(time.Minute))
 
+		It("Should wait until no gardenlet pods exist anymore", func(ctx SpecContext) {
+			Eventually(ctx, func(g Gomega) []corev1.Pod {
 				podList := &corev1.PodList{}
-				g.Expect(shootClient.Client().List(ctx, podList, client.InNamespace("garden"), client.MatchingLabels{"app": "gardener", "role": "gardenlet"})).To(Succeed())
+				g.Expect(s.ShootContext.ShootKomega.List(podList, client.InNamespace(v1beta1constants.GardenNamespace), client.MatchingLabels{"app": "gardener", "role": "gardenlet"})()).To(Succeed())
 				return podList.Items
 			}).WithPolling(5 * time.Second).Should(BeEmpty())
```

Comment:
The following would be more elegant:
```go
Eventually(ctx, s.ShootContext.ShootKomega.List(&corev1.PodList{}, client.InNamespace(v1beta1constants.GardenNamespace), client.MatchingLabels{"app": "gardener", "role": "gardenlet"})).
				WithPolling(5 * time.Second).Should(HaveField("Items", BeEmpty()))
```

Reply:
This doesn't work for me as the Komega List function only returns an error, not the implicitly defined PodList.
Am I missing something?